File: src/main/java/net/thedragonteam/armorplus/commands/SubCommandBase.java
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.commands;

import net.minecraft.command.ICommand;
import net.minecraft.command.ICommandSender;
import net.minecraft.server.MinecraftServer;

import java.util.Locale;

import static net.thedragonteam.armorplus.util.TextUtils.formattedText;
import static net.thedragonteam.armorplus.util.TextUtils.setText;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 */
public class SubCommandBase implements ISubCommand {

    private String subCommandName;
    private ICommand parentCommand;

    public SubCommandBase(ICommand parentCommand, String subCommandName) {
        this.subCommandName = subCommandName;
        this.parentCommand = parentCommand;
    }

    protected String capitalizeFirstLetter(String toCapital) {
        return String.valueOf(toCapital.charAt(0)).toUpperCase(Locale.ENGLISH) + toCapital.substring(1);
    }

    public void displayHelpString(ICommandSender commandSender, String display, Object... info) {
        commandSender.sendMessage(setText(formattedText(display, info)));
    }

    public void displayErrorString(ICommandSender commandSender, String display, Object... info) {
        commandSender.sendMessage(setText(formattedText(display, info)));
    }

    public void displaySuccessString(ICommandSender commandSender, String display, Object... info) {
        commandSender.sendMessage(setText(formattedText(display, info)));
    }

    @Override
    public String getSubCommandName() {
        return subCommandName;
    }

    @Override
    public ICommand getParentCommand() {
        return parentCommand;
    }

    @Override
    public String getArgUsage(ICommandSender commandSender) {
        return "";
    }

    @Override
    public String getHelpText() {
        return "";
    }

    @Override
    public void processSubCommand(MinecraftServer server, ICommandSender commandSender, String[] args) {

    }
}

File: src/main/java/net/thedragonteam/armorplus/commands/SubCommandBase.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.commands

import net.minecraft.command.ICommand
import net.minecraft.command.ICommandSender
import net.minecraft.server.MinecraftServer
import net.thedragonteam.thedragonlib.util.TextUtils.formattedText
import net.thedragonteam.thedragonlib.util.TextUtils.setText
import java.util.*

/**
 * net.thedragonteam.armorplus.commands
 * ArmorPlus created by sokratis12GR on 6/25/2016 9:54 AM.
 * - TheDragonTeam
 */
open class SubCommandBase(override val parentCommand: ICommand, override val subCommandName: String) : ISubCommand {

    override fun getHelpText(): String = ""

    override fun getArgUsage(commandSender: ICommandSender): String = ""

    override fun processSubCommand(server: MinecraftServer, cmdSender: ICommandSender, args: Array<String>) {}

    protected fun capitalizeFirstLetter(toCapital: String): String = toCapital[0].toString().toUpperCase(Locale.ENGLISH) + toCapital.substring(1)

    companion object {

        fun displayHelpString(commandSender: ICommandSender, display: String, vararg info: Any) = commandSender.sendMessage(setText(formattedText(display, *info)))

        fun displayErrorString(commandSender: ICommandSender, display: String, vararg info: Any) = commandSender.sendMessage(setText(formattedText(display, *info)))

        fun displaySuccessString(commandSender: ICommandSender, display: String, vararg info: Any) = commandSender.sendMessage(setText(formattedText(display, *info)))
    }
}